### PR TITLE
PUD-1607: Add Missing Documents to RecallResponse

### PR DESCRIPTION
**Pull request by philsegal**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/RecallControllerTest.kt [hunk 1]
```diff
@@ -579,4 +611,188 @@ class RecallControllerTest {
 
     verify { recallService.confirmRecallType(recallId, request, currentUserId) }
   }
+
+  @Test
+  fun `missingDocuments is null when probationInfo not set`() {
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/RecallControllerTest.kt:616]: Scope and extent of tests here suggest that (for a future iteration?) we might have a `parameterizedTest` for this feature - perhaps via move the `missingDocuments` logic into a collaborator?

**Review by AndyNewton — APPROVED** [review]
Approved ... with a couple of ... you know ... comments.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/RecallController.kt [hunk 2]
```diff
@@ -299,6 +305,27 @@ class RecallController(
...
+      requiredCategories = setOf(PART_A_RECALL_REPORT, LICENCE)
+      desiredCategories = setOf(OASYS_RISK_ASSESSMENT, PREVIOUS_CONVICTIONS_SHEET)
+    }
+    if (recallTypeOrNull() == RecallType.STANDARD && dossierCreatedByUserId != null) {
+      requiredCategories = requiredCategories.plus(PART_B_RISK_REPORT)
+    }
+    val requiredButMissing = determineMissingCategories(requiredCategories)
+    val desiredButMissing = determineMissingCategories(desiredCategories)
+    return if (requiredButMissing.isEmpty() && desiredButMissing.isEmpty())
+      null
```
[the remark this answers is not in the document]
Comment (philsegal) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/RecallController.kt:321]: I kept going back and forth on this. @jonnywyatt - any thoughts before I merge? should it return null? Or an empty object when there is nothing missing?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/RecallControllerTest.kt (hunk 1 above)
Comment (philsegal) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/RecallControllerTest.kt:616]: Yep, I did think about this too. Would make sense at some point to extract

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/RecallController.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (jonnywyatt) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/RecallController.kt:321]: I don't mind either

**Pull request merged**